internals/overlord/checkstate/manager.go

```diff
@@ -349,10 +375,75 @@ type CheckInfo struct {
 type CheckStatus string
 
 const (
-	CheckStatusUp   CheckStatus = "up"
-	CheckStatusDown CheckStatus = "down"
+	CheckStatusUp       CheckStatus = "up"
+	CheckStatusDown     CheckStatus = "down"
+	CheckStatusInactive CheckStatus = "inactive"
 )
 
 type checker interface {
 	check(ctx context.Context) error
 }
+
+// StartChecks starts the specified checks by creating a new perform-checks
+// change for each check that is in the current plan and not already running.
+func (m *CheckManager) StartChecks(currentPlan *plan.Plan, checks []string) ([]string, error) {
+	m.state.Lock()
+	defer m.state.Unlock()
+
+	// If any check specified is not in the plan, return an error.
+	for _, name := range checks {
+		if _, ok := currentPlan.Checks[name]; !ok {
+			return nil, fmt.Errorf("cannot find check %q in plan", name)
+		}
+	}
+	var started []string
+	for _, name := range checks {
+		check := currentPlan.Checks[name] // We know this is ok because we checked it above.
+		info, ok := m.checks[name]
+		if !ok {
+			panic(fmt.Sprintf("check %s is in the plan but not known to the manager", name))
```

Comment:
I suspect this can happen if the plan was just updated? Or can that not happen because of how the updates and the plan-changed listener is called inside a mutex? I'm not sure off the top of my head, just want to make sure we've thought through it.